src/collectors/dirstats/dirstats.py

```diff
@@ -0,0 +1,139 @@
+# coding=utf-8
+
+"""
+Collect given directories stats.
+
+"""
+
+import os
+
+from stat import S_ISDIR, S_ISREG
+from time import time
+
+try:
+    import Queue as queue
+except ImportError:
+    import queue
+
+import diamond.collector
+
+BLOCK = 512
```

Comment:
А размер блока всегда 512?

Reply:
Это значение занесено в константу по двум причинам:

>st_blocks - number of 512-byte blocks allocated for file
(https://docs.python.org/2.7/library/os.html)

>here is no correlation between values of the st_blocks and st_blksize, and the f_bsize (from <sys/statvfs.h>) structure members.
>>Under Linux, 512 turns out to be the right multiplier (and not st_blksize)
(https://bugs.python.org/issue10016)

Т.е. нет какого то более гибкого выражения вроде A*B чтобы посчитать размер файла в абсолютно всех случаях. Можно конечно пользоваться только os.stat(fullpath).st_size, но тогда не будет возможности считать размер sparse-файлов, вроде тех-же графитовских *wsp, размера виртуальных жестких дисков и пр.

Reply:
В доке к python 3.6 уже просто вот так пишут:
>st_blocks
Number of 512-byte blocks allocated for file. This may be smaller than st_size/512 when the file has holes.

Reply:
Понял, ок.